### PR TITLE
Rewrite the rest of math tests

### PR DESCRIPTION
**Pull request by asi1024**

### CODE REVIEW

File: tests/chainerx_tests/unit_tests/routines_tests/test_math.py [hunk 1]
```diff
@@ -1259,19 +1306,12 @@ def func_scalar(self, xp, a, scalar):
...
     ((2, 3, 4), (-2, 2, 0)),
 ])
-@pytest.mark.parametrize_device(['native:0', 'cuda:0'])
-def test_sum(is_module, xp, device, shape, axis, keepdims, dtype):
-    a = array_utils.create_dummy_ndarray(xp, shape, dtype)
-    if is_module:
-        out = xp.sum(a, axis=axis, keepdims=keepdims)
-    else:
-        out = a.sum(axis=axis, keepdims=keepdims)
+class TestSum(UnaryMathTestBase, op_utils.NumpyOpTest):
```
Comment (hvy) [inline tests/chainerx_tests/unit_tests/routines_tests/test_math.py]: How about keeping the `is_module` parameterization?

**Comment by hvy** [comment]
Maximum and AMax seem left out but was it intentional?

File: tests/chainerx_tests/unit_tests/routines_tests/test_math.py (hunk 1 above)
Comment (asi1024) [reply tests/chainerx_tests/unit_tests/routines_tests/test_math.py]: Fixed.

**Comment by asi1024** [comment]
I'll fix `Maximum` test in another PR.

**Comment by hvy** [comment]
Jenkins, test this please.

File: tests/chainerx_tests/unit_tests/routines_tests/test_math.py [hunk 2]
```diff
@@ -1651,12 +1695,9 @@ def test_max_amax():
     assert chainerx.amax is chainerx.max
 
 
-@chainerx.testing.numpy_chainerx_array_equal(
-    accept_error=(ValueError, chainerx.DimensionError), strides_check=False)
-@pytest.mark.parametrize('input,axis', [
+_sum_params = [
```
Comment (hvy) [inline tests/chainerx_tests/unit_tests/routines_tests/test_math.py]: Should it be `_max_params` ?

File: tests/chainerx_tests/unit_tests/routines_tests/test_math.py [hunk 3]
```diff
@@ -1669,35 +1710,71 @@ def test_max_amax():
...
-        return xp.zeros(())  # invalid combination of data and dtype
-
-    a = xp.array(a_np)
-    if is_module:
-        return xp.max(a, axis)
-    else:
-        return a.max(axis)
+@pytest.mark.parametrize('dtype', chainerx.testing.all_dtypes)
+@pytest.mark.parametrize('is_module', [True, False])
+def test_sum_invalid_shapes_and_axis(device, array, axis, dtype, is_module):
```
Comment (hvy) [inline tests/chainerx_tests/unit_tests/routines_tests/test_math.py]: Ditto, this should be "max" I think?

File: tests/chainerx_tests/unit_tests/routines_tests/test_math.py (hunk 2 above)
Comment (hvy) [inline tests/chainerx_tests/unit_tests/routines_tests/test_math.py]: Should it be `_max_params` ?

File: tests/chainerx_tests/unit_tests/routines_tests/test_math.py (hunk 3 above)
Comment (hvy) [inline tests/chainerx_tests/unit_tests/routines_tests/test_math.py]: Ditto, this should be "max" I think?

File: tests/chainerx_tests/unit_tests/routines_tests/test_math.py (hunk 2 above)
Comment (asi1024) [reply tests/chainerx_tests/unit_tests/routines_tests/test_math.py]: Fixed.

File: tests/chainerx_tests/unit_tests/routines_tests/test_math.py (hunk 3 above)
Comment (asi1024) [reply tests/chainerx_tests/unit_tests/routines_tests/test_math.py]: Fixed.

**Comment by hvy** [comment]
Jenkins, test this please.

**Comment by hvy** [comment]
Can you fix this failure? https://jenkins.preferred.jp/job/chainer/job/chainer_pr/185/TEST=CHAINERX_chainerx-py3,label=mn1-p100/console

**Comment by hvy** [comment]
Jenkins, test this please.

**Comment by hvy** [comment]
CI failures seem unrelated. LGTM!

**Pull request merged**